url93da129d40_marioskogias_blkin_hadoop_tgz-pJ8-mythrift_MyZipkinThriftJ8/src/mythrift/Endpoint.java

```diff
@@ -0,0 +1,579 @@
+  }
+
+  /** Returns true if field service_name is set (has been assigned a value) and false otherwise */
+  public boolean isSetService_name() {
+    return this.service_name != null;
+  }
+
+  public void setService_nameIsSet(boolean value) {
+    if (!value) {
+      this.service_name = null;
+    }
+  }
+
+  public void setFieldValue(_Fields field, Object value) {
+    switch (field) {
+    case IPV4:
+      if (value == null) {
+        unsetIpv4();
+      } else {
+        setIpv4((Integer)value);
+      }
+      break;
+
+    case PORT:
+      if (value == null) {
+        unsetPort();
+      } else {
+        setPort((Short)value);
+      }
+      break;
+
+    case SERVICE_NAME:
+      if (value == null) {
+        unsetService_name();
+      } else {
+        setService_name((String)value);
+      }
+      break;
+
+    }
+  }
+
+  public Object getFieldValue(_Fields field) {
+    switch (field) {
+    case IPV4:
+      return Integer.valueOf(getIpv4());
+
+    case PORT:
+      return Short.valueOf(getPort());
+
+    case SERVICE_NAME:
+      return getService_name();
+
+    }
+    throw new IllegalStateException();
+  }
+
+  /** Returns true if field corresponding to fieldID is set (has been assigned a value) and false otherwise */
+  public boolean isSet(_Fields field) {
+    if (field == null) {
+      throw new IllegalArgumentException();
+    }
+
+    switch (field) {
+    case IPV4:
+      return isSetIpv4();
+    case PORT:
+      return isSetPort();
+    case SERVICE_NAME:
+      return isSetService_name();
+    }
+    throw new IllegalStateException();
+  }
+
+  @Override
+  public boolean equals(Object that) {
+    if (that == null)
+      return false;
+    if (that instanceof Endpoint)
+      return this.equals((Endpoint)that);
+    return false;
+  }
+
+  public boolean equals(Endpoint that) {
+    if (that == null)
+      return false;
+
+    boolean this_present_ipv4 = true;
+    boolean that_present_ipv4 = true;
+    if (this_present_ipv4 || that_present_ipv4) {
+      if (!(this_present_ipv4 && that_present_ipv4))
+        return false;
+      if (this.ipv4 != that.ipv4)
+        return false;
+    }
+
+    boolean this_present_port = true;
+    boolean that_present_port = true;
+    if (this_present_port || that_present_port) {
+      if (!(this_present_port && that_present_port))
+        return false;
+      if (this.port != that.port)
+        return false;
+    }
+
+    boolean this_present_service_name = true && this.isSetService_name();
+    boolean that_present_service_name = true && that.isSetService_name();
+    if (this_present_service_name || that_present_service_name) {
+      if (!(this_present_service_name && that_present_service_name))
+        return false;
+      if (!this.service_name.equals(that.service_name))
+        return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    return 0;
+  }
+
+  @Override
+  public int compareTo(Endpoint other) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.